Backend/src/main/java/com/onbank/api/model/Transfer.java

```diff
@@ -1,38 +1,59 @@
 package com.onbank.api.model;
 
 import lombok.Data;
-import lombok.Getter;
-import lombok.Setter;
 
 import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Positive;
 import javax.validation.constraints.Size;
 import java.math.BigDecimal;
 import java.time.LocalDateTime;
+import javax.validation.constraints.Pattern;
 
 @Data
 @Entity
 @Table(name = "transfer")
 public class Transfer {
     @Id
-    @GeneratedValue(strategy = GenerationType.IDENTITY)
     @Column(name="Id", nullable=false, unique = true)
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     @Column(name="date", nullable=false)
+    @NotNull(message = "Date cannot be empty.")
     private LocalDateTime date;
     @Column(name="name", nullable=false, length=50)
+    @NotNull(message = "Name cannot be empty.")
     private String name;
     @Column(name="surname", nullable=false, length=100)
+    @NotNull(message = "Surname cannot be empty.")
     private String surname;
     @Column(name="accountnumber", nullable=false)
-    @Size(min = 26, max = 28)
+    @Size(min = 26, max = 28, message = "Account number must be between 26-28 characters.")
+    @NotNull(message = "Account number cannot be empty.")
+    @Pattern(regexp = "(^[A-Z]{2}\\d{26}$)|(^\\d{26}$)")
```

Comment:
Wydaje mi się że te adnotacje javax.validation.constraints nie zadziałają ne encjach, w sensie one z tego co pamiętam wywołują się tylko przy wywołaniu adnotacji @Valid tak jak to zrobiłeś z dto. Możesz zweryfikować, usunąć na chwilę z DTO jakąś walidację np numeru konta i zobaczyć czy zwaliduje przy zapisie. Jeżeli nie to można usunąć z encji te wszystkie walidatory.

Reply:
Sprawdziłem, walidacja w samych encjach też działa. Pytanie czy jest mimo to potrzebna? Skoro DTO'sy to załatwią.